### PR TITLE
fix(eslint): Restrict the rules and fix errors

### PR DESCRIPTION
**Pull request by zharinov**

### CODE REVIEW

File: lib/workers/pr/body/index.ts [hunk 1]
```diff
@@ -13,7 +13,9 @@ import { getPrUpdatesTable } from './updates-table';
 function massageUpdateMetadata(config: BranchConfig): void {
   config.upgrades.forEach((upgrade) => {
     /* eslint-disable no-param-reassign */
-    const { homepage, sourceUrl, sourceDirectory, changelogUrl } = upgrade;
+    const { sourceUrl, sourceDirectory } = upgrade;
+    const homepage = upgrade.homepage as string;
+    const changelogUrl = upgrade.changelogUrl as string;
```
Comment (viceice) [inline lib/workers/pr/body/index.ts:18]: Maybe we should proper type them on `BranchUpgrade`  🤔

File: lib/workers/repository/onboarding/pr/index.ts [hunk 2]
```diff
@@ -89,11 +89,13 @@ If you need any further assistance then you can also [request help here](${confi
   prBody = prBody.replace('{{PRLIST}}\n', getPrList(config, branches));
   // istanbul ignore if
   if (config.prHeader) {
-    prBody = (config.prHeader || '') + '\n\n' + prBody;
+    const prHeader = String(config.prHeader || '');
+    prBody = `${prHeader}\n\n${prBody}`;
```
Comment (viceice) [inline lib/workers/repository/onboarding/pr/index.ts:93]: should be fixed later, as `prHeader` and `prFooter` should be string.

File: lib/workers/pr/body/index.ts (hunk 1 above)
Comment (zharinov) [reply lib/workers/pr/body/index.ts]: Just added the fields to `BranchUpgradeConfig`

File: lib/workers/pr/body/index.ts [hunk 3]
```diff
@@ -13,7 +13,9 @@ import { getPrUpdatesTable } from './updates-table';
 function massageUpdateMetadata(config: BranchConfig): void {
   config.upgrades.forEach((upgrade) => {
     /* eslint-disable no-param-reassign */
-    const { homepage, sourceUrl, sourceDirectory, changelogUrl } = upgrade;
+    const { sourceUrl, sourceDirectory } = upgrade;
+    const homepage = upgrade.homepage;
+    const changelogUrl = upgrade.changelogUrl;
```
Comment (rarkins) [inline lib/workers/pr/body/index.ts:18]: Why do these changes help?

Comment (zharinov) [reply lib/workers/pr/body/index.ts:18]: `BranchConfig` -> `RenovateConfig` -> `Record<string, unknown>`

Comment (rarkins) [reply lib/workers/pr/body/index.ts:18]: I still don't get why the change in syntax affects anything.

Comment (zharinov) [reply lib/workers/pr/body/index.ts:18]: Ah, now we can revert it

File: lib/workers/repository/onboarding/pr/index.ts (hunk 2 above)
Comment (rarkins) [reply lib/workers/repository/onboarding/pr/index.ts:93]: @viceice can you clarify?

Comment (viceice) [reply lib/workers/repository/onboarding/pr/index.ts:93]: they should be defined on the type as string, then the `String(...)` wrapper shouldn't needed

**Comment by viceice** [comment]
@zharinov You should request another review if this is ready again 😉

**Review by viceice — DISMISSED** [review]

**Comment by zharinov** [comment]
Well, if checks are still passing, I guess it's up-to-date for now

**Pull request merged**